Objects/typeobject.c

```diff
@@ -2241,6 +2241,12 @@ extra_ivars(PyTypeObject *type, PyTypeObject *base)
         type->tp_weaklistoffset + sizeof(PyObject *) == t_size &&
         type->tp_flags & Py_TPFLAGS_HEAPTYPE)
         t_size -= sizeof(PyObject *);
+    if (!(type->tp_flags & Py_TPFLAGS_MANAGED_DICT) &&
```

Comment:
Is is correct to check `Py_TPFLAGS_MANAGED_DICT` during class creation. It is set during class creation.
https://github.com/python/cpython/blob/main/Objects/typeobject.c#L2989

Reply:
I think setting the flag happens before the MRO check. https://github.com/python/cpython/blob/main/Objects/typeobject.c#L3174-L3186
(`type_new_descriptors` then `PyType_Ready` ). 
(`PyType_Ready` -> `type_ready` -> `type_ready_mro`)

Reply:
If you factor out the check into a macro, the code would be a bit clearer.
```C
#define FIELD_AT_OFFSET(name, offset) type->tp_ ## name && base->tp_## name == 0 ...
```